packages/editor-ui/src/views/NodeView.v2.vue

```diff
@@ -910,8 +911,101 @@ function trackRunWorkflowToNode(node: INodeUi) {
 	void externalHooks.run('nodeView.onRunNode', telemetryPayload);
 }
 
-async function openExecution(_executionId: string) {
-	// @TODO
+async function openExecution(executionId: string) {
+	canvasStore.startLoading();
+	resetWorkspace();
+
+	let data: IExecutionResponse | undefined;
+	try {
+		data = await workflowsStore.getExecution(executionId);
+	} catch (error) {
+		toast.showError(error, i18n.baseText('nodeView.showError.openExecution.title'));
+		return;
+	}
+	if (data === undefined) {
+		throw new Error(`Execution with id "${executionId}" could not be found!`);
+	}
+
+	await initializeData();
+	await initializeWorkspace(data.workflowData);
+	workflowsStore.setWorkflowExecutionData(data);
```

Comment:
Why is all this logic in `openExecution` and `onExecutionOpenedWithError` needed? The execution itself is displayed in an iframe so why do we need to call `initializeWorkspace` etc?

In a perfect world™️, should the executions view handle all this stuff? and NodeView just do a router navigation?

Reply:
Technically, if there would be no iframe, we'd be able to do this inside of the executions view. This is basically us hacking our way around it. 

I'm adding a task so we remember to clean this up after we remove the need for an iframe.